src/vivarium/framework/randomness.py

```diff
@@ -627,10 +611,10 @@ def setup(self, builder):
 
         self.resources = builder.resources
         self._add_constraint = builder.lifecycle.add_constraint
-        builder.lifecycle.add_constraint(self.get_randomness_stream, allow_during=['setup'])
-        builder.lifecycle.add_constraint(self.register_simulants,
-                                         restrict_during=['initialization', 'setup', 'post_setup',
-                                                          'simulation_end', 'report'])
+        self._add_constraint(self.get_seed, restrict_during=['initialization'])
```

Comment:
Can we extract these stage names as constants so we don't have strings everywhere?

Reply:
Yes, but smartly.  Make a ticket in the next sprint backlog to survey string constants across vivarium and vph and then we'll come up with a strategy for handling categories of string constants.

Reply:
https://jira.ihme.washington.edu/browse/MIC-1615